src/pfe/file-watcher/server/test/functional-test/functional.test.ts

```diff
@@ -8,23 +8,33 @@
  * Contributors:
  *     IBM Corporation - initial API and implementation
  *******************************************************************************/
+import mocha from "mocha";
 import { expect } from "chai";
+import fs from "fs";
+import path from "path";
 
 import * as utils from "./lib/utils";
 // set all the test environmental variables
 utils.setTestEnvVariables();
 import * as app_configs from "./configs/app.config";
-import * as git_configs from "./configs/github.config";
-import { localeTestSuite } from "./tests/locale.test";
-import { createTestSuite } from "./tests/create.test";
+import * as pfe_configs from "./configs/pfe.config";
+import * as timoutConfigs from "./configs/timeout.config";
 
-import { ProjectCreation } from "./lib/project-creation";
+import GenericTestSuite from "./suites/generic/index";
+import ProjectTestSuite from "./suites/project/index";
+
+import { ProjectCreation } from "./lib/project";
+import { SocketIO } from "./lib/socket-io";
 
 const projectTypes = app_configs.projectTypes;
 const dockerProjects = app_configs.supportedDockerProjects;
 const projectConfigs = app_configs.projectConfigs;
 
-describe("Functional Test Suite", () => {
+const socket = new SocketIO();
+const genericSuite = new GenericTestSuite(socket);
+const projectSuite = new ProjectTestSuite(socket);
+
+describe("PFE - functional test", () => {
   beforeEach("check for pfe to be up", (done) => {
```

Comment:
Is it necessary to check for PFE to be up for each test case? Currently we have ~200 test cases, seem like we check ~200 times for checking PFE to be up. It will increase the overall testing time as the test cases grow

Reply:
We don't check 200 times. This is a before hook for the whole test. Only checks once.

Reply:
This is what I got:

![Screen Shot 2019-08-09 at 5 17 35 PM](https://user-images.githubusercontent.com/15131537/62809692-b3d84a00-bac9-11e9-94f4-e331956dbf69.png)

Reply:
Actually sorry. Yes forgot it was a `beforeEach` hook. Now that I remember, we need to check that PFE is up before each test to verify that it didn't go down. Because our tests can't run without PFE being up.

Reply:
Then we go back the original question, is it necessary to check for PFE to be up for ~200 times? I think check before each test suite would be sufficient, it's same idea for our Microclimate test automation, we don't check that for each test case.

Reply:
I think it is needed. Because test cases in a suite depend on PFE to be up and if suddenly PFE goes down our test will be stuck on timeouts. With the simple GET request we can easily check if PFE is up before every test. It is not an overhead IMO.

Reply:
Actually our test cases are running inside PFE container, if PFE goes down our tests will terminates immediately and there is no timeout issue for PFE goes down. Given that, I don't think we need to check if the PFE container is up or not, test cases themselves can help us.

Reply:
That's actually true. Good catch. We had this implementation before when we were running before outside container so it made sense. But now if PFE goes down, you are right the test will also go down. So it's not needed anymore